src/libraries/System.Private.Xml/src/System/Xml/XmlCharType.cs

```diff
@@ -39,91 +41,42 @@ internal unsafe struct XmlCharType
         // bitmap for public ID characters - 1 bit per character 0x0 - 0x80; no character > 0x80 is a PUBLIC ID char
         private const string s_PublicIdBitmap = "\u2400\u0000\uffbb\uafff\uffff\u87ff\ufffe\u07ff";
 
-        // size of XmlCharType table
-        private const uint CharPropertiesSize = (uint)char.MaxValue + 1;
 
-        // static lock for XmlCharType class
-        private static object? s_Lock;
+        public static XmlCharType Instance => default;
```

Comment:
I still think it would be nice to get rid of this method and make all methods static. It is a trivial find&replace operation, with a few manual edits.

Reply:
Changed XmlCharType to a static class with all static methods, removed all instantiations and fixed up all call sites. I also cleaned up the naming since I was ending up making many changes to the file.